cmd/promtool/rules.go

```diff
@@ -105,7 +105,7 @@ func (importer *ruleImporter) importRule(ctx context.Context, ruleExpr, ruleName
 		endOfBlock := startOfBlock + blockDuration - 1
 
 		currStart := max(startOfBlock/int64(time.Second/time.Millisecond), start.Unix())
-		startWithAlignment := grp.EvalTimestamp(time.Unix(currStart, 0).UTC().UnixNano())
+		startWithAlignment := grp.EvalTimestamp(time.Unix(currStart, 0).UTC())
```

Comment:
How can this fix it ? UTC removed the monotonic reading anyway.

Reply:
There is no need for monotonic clock in promtool. Is is only for the rules manager.
promtool does not measure time.